### ISSUE TITLE
Error: code: engine.platform or engine.scriptSrc is not defined in custom engine "cordova-browser" from plugin "phonegap-plugin-push" for android warn

### ISSUE THREAD

**Issue by TMMeijers** [phonegap/phonegap-plugin-push#2235]
### Expected Behaviour

When running `cordova platform add android --nofetch` it installs the plugin just like it did when on plugin version 2.1.2.

### Actual Behaviour

The following warning/error comes up
```Fetching plugin "phonegap-plugin-push@^2.2.0" via npm
Installing "phonegap-plugin-push" for android
Failed to restore plugin "phonegap-plugin-push" from config.xml. You might need to try adding it again. Error: code: engine.platform or engine.scriptSrc is not defined in custom engine "cordova-browser" from plugin "phonegap-plugin-push" for android warn
```

### Reproduce Scenario (including but not limited to)

The following is in the config.xml in case it is needed: 
```
<platform name="android">
    <resource-file src="google-services.json" target="google-services.json" />
</platform>
<plugin name="phonegap-plugin-push" spec="^2.2.0">
    <variable name="FCM_VERSION" value="11.0.1" />
</plugin>
```

#### Steps to Reproduce 

Remove all plugins and platforms and run `cordova platform add android --nofetch` with versions specified below.

#### Cordova CLI version and cordova platform version

    cordova 7.1
    cordova-android 6.4

#### Plugin version

    2.2.0

**Comment by ntamas** [comment]
Removing `<engine name="cordova-browser" version=">=5.0.3" />` from `plugin.xml` seems to solve the issue, but I'm not familiar enough with `plugin.xml` so I don't know whether this is the best solution or only a temporary workaround.

**Comment by TMMeijers** [comment]
It would seem to me that every new install or update of the plugin would revert that change so it would definitely be a temporary work-around and also not feasible in an automated build process.

**Comment by gehirnmitei** [comment]
same Problem here

**Comment by lesteenman** [comment]
This happens to me when I'm using cordova-android 7.1.0 and phonegap-plugin-push 2.2.0, but not with cordova-android 7.0.0 using plugin version 2.1.3.

**Comment by macdonst** [comment]
First, this is a duplicate of #2233.

Second, this is expected behaviour. Version 2.2.0 needs cordova-android 7.1.0 or better and the reporter is using cordova-android 6.4.0.

Third, this is an important bump of cordova-android as it fixes issue #2051 which may cause the plugin to fail altogether on certain versions of Android.

If you don't need something in the 2.2.0 release just don't upgrade.

**Comment by lesteenman** [comment]
I will test again later today, but I did get this issue with cordova-android 7.1.0 and the 2.2.0 version of the plugin.

**Comment by macdonst** [comment]
@lesteenman okay, please let me know when you get a chance to re-test. Also, let me know what version of the cordova-cli you are using when you report back.

**Comment by macdonst** [comment]
Mea culpa everyone. Apparently my workspace had a development version of cordova linked that supports limiting the version of cordova-browser. I have released a 2.2.1 to fix the issue. Please let me know if it is not resolved.

**Comment by lesteenman** [comment]
It works now. Thanks!
